### PR TITLE
updated Actor How To docs for reminder intervals

### LINKED ISSUE #1600: Actors reminder and triggers support ISO 8601 intervals standard format
This is to document https://github.com/dapr/dapr/pull/3095 (and dup https://github.com/dapr/dapr/issues/2513)
Update the Reference Documentation here https://docs.dapr.io/reference/api/actors_api/#create-actor-reminder and https://docs.dapr.io/reference/api/actors_api/#create-actor-timer

This also can be documented in the SDKs.

### PR DESCRIPTION
**Pull request by jigargandhi**
Thank you for helping make the Dapr documentation better!
Reference issue: #1600 
**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

### CODE REVIEW

**Comment by jigargandhi** [comment]
I think this needs update in the https://docs.dapr.io/reference/api/actors_api/#create-actor-reminder section too. I will update this PR with changes to api reference too.

**Comment by orizohar** [comment]
> I think this needs update in the https://docs.dapr.io/reference/api/actors_api/#create-actor-reminder section too. I will update this PR with changes to api reference too.

Thanks, please do

**Comment by jigargandhi** [comment]
> > I think this needs update in the https://docs.dapr.io/reference/api/actors_api/#create-actor-reminder section too. I will update this PR with changes to api reference too.
> 
> Thanks, please do

Done

File: daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ Refer [api spec]({{< ref "actors_api.md#invoke-timer" >}}) for more details.
 
 ### Actor reminders
 
-Reminders are a mechanism to trigger *persistent* callbacks on an actor at specified times. Their functionality is similar to timers. But unlike timers, reminders are triggered under all circumstances until the actor explicitly unregisters them or the actor is explicitly deleted. Specifically, reminders are triggered across actor deactivations and failovers because the Dapr actors runtime persists the information about the actors' reminders using Dapr actor state provider.
+Reminders are a mechanism to trigger *persistent* callbacks on an actor at specified times. Their functionality is similar to timers. But unlike timers, reminders are triggered under all circumstances until the actor explicitly unregisters them or the actor is explicitly deleted or the number of times reminders can invoke are exhausted. Specifically, reminders are triggered across actor deactivations and failovers because the Dapr actors runtime persists the information about the actors' reminders using Dapr actor state provider.
```
Comment (artursouza) [inline daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md]: "or the number in invocations is exhausted."

File: daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md [hunk 2]
```diff
@@ -111,6 +111,28 @@ The following request body configures a reminder with a `dueTime` 15 seconds and
...
+```json
+{
+  "dueTime":"0h0m0s0ms",
+  "period":"P0Y0M0W0DT0H0M15S"
+}
+```
+The designators for zero are optional and the above `period` can be simplified to `PT15S`.
+ISO 8601 specifies multiple recurrence formats but only the duration format is currently supported.
+
+#### Reminders with repetitions
```
Comment (artursouza) [inline daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md:124]: @dmitsh Please, see if this overlaps with your change.

**Review by artursouza — CHANGES_REQUESTED** [review]
minor change request.

File: daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md (hunk 1 above)
Comment (jigargandhi) [reply daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md]: Updated

**Comment by dmitsh** [comment]
I'm working on a related issue https://github.com/dapr/dapr/issues/2906, scheduled for v1.4 release.
I'm planning to add ISO format to timers, so we would have similar syntax for setting both timers and reminders.

**Comment by berndverst** [comment]
@dmitsh @jigargandhi please make sure we explain that only this *single* type of ISO 8601 interval formats is supported.

There are at least 4 different formats but we don't support all.

We need to be explicit about the limitations or else someone will encounter problems in the future.

File: daprdocs/content/en/reference/api/actors_api.md [hunk 3]
```diff
@@ -178,7 +178,15 @@ Creates a persistent reminder for an actor.
...
 ```
 
-Body:
+#### Request Body
+
+A JSON object with the following fields
+Field | Description
+---|---
+dueTime | specifies time after which the reminder is invoked, its format should be [time.ParseDuration](https://pkg.go.dev/time#ParseDuration) format
+period | specifies the period between different invocations, its format should be [time.ParseDuration](https://pkg.go.dev/time#ParseDuration) format or ISO 8601 duration format with optional recurrence.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/reference/api/actors_api.md]:
```suggestion
A JSON object with the following fields:

| Field | Description |
|-------|--------------|
| dueTime | Specifies the time after which the reminder is invoked, its format should be [time.ParseDuration](https://pkg.go.dev/time#ParseDuration) format
| period | Specifies the period between different invocations, its format should be [time.ParseDuration](https://pkg.go.dev/time#ParseDuration) format or ISO 8601 duration format with optional recurrence.
```

Comment (AaronCrawfis) [reply daprdocs/content/en/reference/api/actors_api.md]: Also please provide an example column

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

**Comment by berndverst** [comment]
@AaronCrawfis I also need to call out that ISO 8601 has 3 other valid formats (these specify start, end or both) but are not supported here (and they cannot be supported).

However those are valid ISO 8601 interval formats. It would be confusing for someone to see ISO 8601 support only to realize that not all ISO 8601 strings actually work.

It needs to be called out in the documentation that the following is not supported:
- start datetime
- end datetime

**Comment by dmitsh** [comment]
> * infinite recurrence (ISO 8601 lets you specify this too)
> * start datetime
> * end datetime

The start and end times in ISO 8601 are specified in the absolute values, which is not applicable in our case.
IMO we just need to clarify in documentation that we only support time intervals with repetitions without start/end datetime.

**Comment by berndverst** [comment]
Yes that's exactly what I'm saying @dmitsh we should do somehow

**Comment by msfussell** [comment]
@berndverst @dmitsh - Could one of you suggest the changes to this PR? We need to get this in today

**Comment by dmitsh** [comment]
@msfussell , IMO no changes are needed. The document clearly states that "ISO 8601 specifies multiple recurrence formats but only the duration format is currently supported", followed by the example.
I'm working on a similar PR for timers, and I will submit another doc PR later to reflect the changes. I might touch some of the reminders code, in which cace I will update the docs accordingly.

**Comment by dmitsh** [comment]
From this doc: "Only duration format of ISO 8601 duration `Rn/PnYnMnWnDTnHnMnS` is supported for `period`."

File: daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md [hunk 4]
```diff
@@ -111,6 +111,28 @@ The following request body configures a reminder with a `dueTime` 15 seconds and
 }
 ```
 
+[ISO 8601 duration](https://en.wikipedia.org/wiki/ISO_8601#Durations) can also be used to specify `period`. The following request body configures a reminder with a `dueTime` 0 seconds an `period` of 15 seconds.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/actors/howto-actors.md:114]: I would add a link here to the community video on this work. 
This is the link https://www.youtube.com/watch?v=B_vkXqptpXY&t=1002s at the right time spot. 

Something like "Watch is video for more information on ISO861 for Reminders"

**Review by orizohar — APPROVED** [review]

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**